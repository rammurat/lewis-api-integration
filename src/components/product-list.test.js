// @ts-nocheck
import 'jsdom-global/register';
import React from 'react'
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux'
import List from './product-list'

import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

const store = {
    subscribe: () => { },
    dispatch: () => { },
    getState: () => ({
        products: {
            initialConfig: {
                isLoading: false
            }
        }
    })
}

it('renders correctly', () => {
    const tree = renderer
        .create(<Provider store={store}><List /></Provider>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

test('check for product list container', () => {
    const wrapper = mount(
        <Provider store={store}>
            <List />
        </Provider>
    )
    expect(wrapper.find('.product-list').length).toBe(1)
})