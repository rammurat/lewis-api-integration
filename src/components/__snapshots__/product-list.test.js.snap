// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="product-list"
>
  <div
    className="row justify-content-end"
  />
</div>
`;
